bcgov/PIMS
# PIMS-385 Add Postgres Container Locally

## 🎯 Summary

  
[PIMS-385](https://citz-imb.atlassian.net/browse/PIMS-385?atlOrigin=eyJpIjoiY2I4OGFmNzdkOGNkNDI5OWI5NDdiODk3MWNiYTIyODciLCJwIjoiaiJ9)


## Changes
Added a Docker container for Postgres. This can be used for local development with the Express API. 

Chose version 13.11 because that's the version running in OpenShift.
Using alpine3.17 image because it had the fewest vulnerabilities (1 high).

## Testing
1. Populate the Postgres env variables based on the .env-template file in root. I just used `postgres` for everything.
2. Start the container with `docker-compose up postgres -d` and wait for the build to complete
3. Check Docker Desktop to see successful messages in logs
<img width="930" alt="image" src="https://github.com/bcgov/PIMS/assets/37922247/b0ae7188-8232-430b-808e-5fccf8689de5">

4. Optional. Connect to Postgres using some tool, such as the SQL Tools extension
<img width="384" alt="image" src="https://github.com/bcgov/PIMS/assets/37922247/ea8106a8-4f0e-4362-bd14-5d7d603541e0">



## 🔰 Checklist


> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.
